library/common/extensions/filters/http/platform_bridge/filter.cc

```diff
@@ -81,20 +82,59 @@ Http::FilterHeadersStatus PlatformBridgeFilter::onHeaders(Http::HeaderMap& heade
 }
 
 Http::FilterDataStatus PlatformBridgeFilter::onData(Buffer::Instance& data, bool end_stream,
+                                                    Buffer::Instance* internal_buffer,
                                                     envoy_filter_on_data_f on_data) {
   // Allow nullptr to act as no-op.
   if (on_data == nullptr) {
     return Http::FilterDataStatus::Continue;
   }
 
-  envoy_data in_data = Buffer::Utility::toBridgeData(data);
+  envoy_data in_data;
+
+  if (iteration_mode_ == IterationMode::Stopped && internal_buffer &&
+      internal_buffer->length() > 0) {
+    // Pre-emptively buffer data to present aggregate to platform.
+    internal_buffer->move(data);
+    in_data = Buffer::Utility::copyToBridgeData(*internal_buffer);
+  } else {
+    in_data = Buffer::Utility::copyToBridgeData(data);
+  }
+
   envoy_filter_data_status result = on_data(in_data, end_stream, platform_filter_.instance_context);
   Http::FilterDataStatus status = static_cast<Http::FilterDataStatus>(result.status);
+  switch (status) {
+  case Http::FilterDataStatus::Continue:
+    if (iteration_mode_ == IterationMode::Stopped) {
+      // Error: iteration must be Resumed first.
```

Comment:
Are we planning to add an error here or will we silently fail?

Reply:
Options are to fail the request or crash the app. Neither seems terribly ideal, but perhaps neither is silently failing. I hadn't made up my mind here yet.